manufacturing-server/src/main.rs

```diff
@@ -268,7 +274,49 @@ async fn main() -> Result<()> {
 
     // Initialize handlers
     let hello = warp::get().map(|| "Hello from the manufacturing server");
-    let handler_ping = fdo_http_wrapper::server::ping_handler();
+    let ud = user_data.clone();
+    let handler_export = warp::get()
+        .and(warp::path("export").map(move || (ud.clone())).and_then(
+            |ud: Arc<ManufacturingServiceUD>| async move {
+                match ud.ownership_voucher_store.load_all_data().await {
+                    Ok(ovs) => Ok(ovs),
+                    Err(_) => Err(Rejection::from(fdo_http_wrapper::server::Error::new(
+                        ErrorCode::InternalServerError,
+                        fdo_data_formats::constants::MessageType::Invalid,
+                        "Error loading ownership vouchers",
+                    ))),
+                }
+            },
+        ))
+        .map(|ovs: Vec<OwnershipVoucher>| {
+            let tmp_dir = TempDir::new("manufacturer-server-ovs").unwrap();
+            for ov in ovs {
+                let file_path = tmp_dir.path().join(ov.header().guid().to_string());
+                let tmp_file = File::create(file_path).unwrap();
+                OwnershipVoucher::serialize_to_writer(&ov, &tmp_file).unwrap();
+            }
```

Comment:
Shouldn't we return 404 if there are no OVs?

Reply:
do you think it's better a 404 or a 500?

Reply:
yeah, probably 404